safe-global/safe-wallet-web
# feat: Staking banner

Show banners for staking on the dashboard and the assets page if the user holds >= 32 ETH.

## What it solves

Resolves [SW-191](https://www.notion.so/safe-global/Dashboard-banner-staking-84cb882dc4eb456a91a01ec19771fbdf?pvs=4)

## How this PR fixes it

This pull request adds a new Staking banner as a replacement of the Swap banner. The changes include implementing the StakingBanner component, integrating it into the dashboard and assets pages, and updating the styles and hooks accordingly.

### Staking Banner Implementation:

* [`src/components/dashboard/StakingBanner/index.tsx`](diffhunk://#diff-b79df4b94dde2ad15a3cbb6b6f50a36c2208b9e5b39c757d90c795f010591a9eR1-R187): Added a new `StakingBanner` component with support for both large and small views. This component includes logic to check for sufficient funds (>= 32 ETH), dark mode, and the possibility to hide it.
* [`src/components/dashboard/StakingBanner/styles.module.css`](diffhunk://#diff-787e7ff2e8bfe9a889e2daba7dbb9fb7a2f523fd4814177285830a266efb4e17R1-R71): Added CSS styles for the `StakingBanner` component, including responsive design and dark mode support.
* [`src/features/stake/hooks/useIsStakingBannerEnabled.ts`](diffhunk://#diff-81fee138ea5d7f32c01366f4a77f4ffdecce22af0c324f78fe6d8c3f568a3e88R1-R10): Created a new hook `useIsStakingBannerEnabled` to determine if the staking banner feature is enabled.

### Integration:

* [`src/components/dashboard/index.tsx`](diffhunk://#diff-55f42e0427f40a290cb06611c4740215af6e5ed241e4f4ade08a3e118e4ccb5bR13-R28): Integrated the `StakingBanner` component into the dashboard, replacing the Swap Widget. [[1]](diffhunk://#diff-55f42e0427f40a290cb06611c4740215af6e5ed241e4f4ade08a3e118e4ccb5bR13-R28) [[2]](diffhunk://#diff-55f42e0427f40a290cb06611c4740215af6e5ed241e4f4ade08a3e118e4ccb5bL50-R52)
* [`src/pages/balances/index.tsx`](diffhunk://#diff-596d293c4682193c77d01b63d368328c1ab7e90113ff4449e832f0b296749070R14-R22): Added the `StakingBanner` component to the assets page. [[1]](diffhunk://#diff-596d293c4682193c77d01b63d368328c1ab7e90113ff4449e832f0b296749070R14-R22) [[2]](diffhunk://#diff-596d293c4682193c77d01b63d368328c1ab7e90113ff4449e832f0b296749070R37-R42)

### Removal of Swap Widget:

* [`src/features/swap/components/SwapWidget/index.tsx`](diffhunk://#diff-7583e5e78f83d957f008d018e0fed9b24efa9f5d1793e2e5f8b2a743f6ff036eL1-L82): Removed the `SwapWidget` component and its associated logic.

### Additional Updates:

* [`src/services/analytics/events/overview.ts`](diffhunk://#diff-18f4f5d59bbd695f538ef8becbd5472e609fd46d9caab6dcffcf0a74d0ca7da5R177-R188): Added new analytics events for the staking banner interactions.
* [`src/utils/chains.ts`](diffhunk://#diff-dbfb6276b90eb214a47841c8615027bc8b39d52b5e0ec4156793901c209673a3R37): Added `STAKING_BANNER` to the `FEATURES` enum.

## How to test it

Load the app for an account that has >= 32 ETH and visit the dashboard and assets pages where the banner should appear at the top. Alternatively, update the `MIN_NATIVE_TOKEN_BALANCE` constant in the `StakingBanner` component.

## Screenshots

### Dashboard
<img width="1581" alt="Screenshot 2024-10-21 at 09 53 52" src="https://github.com/user-attachments/assets/7b052bbb-738e-4829-aa08-4203ffa4ff63">
<img width="1569" alt="Screenshot 2024-10-21 at 09 53 46" src="https://github.com/user-attachments/assets/8a6e874c-4ac1-457a-ace1-74f036dc3f30">
<img width="551" alt="Screenshot 2024-10-21 at 09 54 03" src="https://github.com/user-attachments/assets/7facdcd6-0ecf-48df-a442-b41cfb4c32cd">

### Assets
<img width="1537" alt="Screenshot 2024-10-23 at 14 20 12" src="https://github.com/user-attachments/assets/b9cc93af-bf5b-416e-b8c1-11d636ce031c">
<img width="1535" alt="Screenshot 2024-10-23 at 14 20 19" src="https://github.com/user-attachments/assets/440f90b9-8c1f-4fa5-9fb2-2e5d067f1fb7">
<img width="590" alt="Screenshot 2024-10-23 at 14 20 35" src="https://github.com/user-attachments/assets/8ba3b9bb-feac-492d-9b30-e1da4ab5f68e">

## Checklist
